[비동기 로그인, 댓글 기능] 에헴 미션 제출합니다.

hyogenie-v opened the pull request:
안녕하세요, 에헴입니다!

비동기로 변경한 기능은
1. 로그인
2. 댓글 작성
3. 댓글 수정

세 가지입니다.
리뷰 잘 부탁드립니당
감사합니다!!

hyogenie-v committed:
feat: ajax로 비동기식 로그인 처리 기능을 추가

1. login.html에 로그인 버튼 이벤트 스크립트를 추가한다
2. 로그인 관련 예외가 발생한 경우에 대한 핸들러의 리턴을 ResponseEntity으로 수정한다
3. 로그 설정 파일인 logback-access.xml를 추가한다

hyogenie-v committed:
feat: ajax로 비동기식 댓글 등록 기능 구현

1. article.html 뷰에 댓글 등록 스크립트를 추가한다
2. CommentController의 댓글 등록 메서드가 ResponseEntity를 리턴하도록 수정한다

hyogenie-v committed:
fix: ajax로 추가한 댓글의 수정 버튼을 클릭해도 뷰가 바뀌지 않는 버그 수정

hyogenie-v committed:
feat: ajax로 비동기식 댓글 수정 기능을 추가

1. article.html에 수정 버튼 이벤트 스크립트를 추가한다
2. CommentController의 댓글 수정 메서드가 ResponseEntity를 리턴하도록 수정한다

hyogenie-v committed:
test: 로그인 및 댓글 등록/수정 기능을 비동기식 통신으로 바꿈에 따른 테스트 수정

hyogenie-v committed:
logs: RollingFileAppender.xml 추가

hyogenie-v committed:
docs: 쉘 스크립트 파일 추가

jihan805 reviewed (changes requested):
안녕하세요 에헴 ~!
구현 잘하셨어요 👍 
몇가지 피드백 남겼으니 확인해주세요 :)

hyogenie-v committed:
feat: 엔티티 클래스에 @Column 애너테이션 추가

1. 엔티티 클래스에 @Column 애너테이션을 추가하여 칼 속성을 지정한다
2. 엔티티의 공통 부분을 BaseEntity 클래스로 추출하여 이를 상속하도록 수정한다
3. BaseEntity 클래스에 생성 날짜 및 최근 수정 날짜 필드를 생성한다

hyogenie-v committed:
feat: 댓글 관련 url을 자원을 표현할 수 있는 구조로 수정

1. CommentController의 경로 매핑을 수정한다
2. 테스트 코드를 수정한다

hyogenie-v committed:
refactor: 패키지 구조 변경

1. 도메인별로 나뉘어져 있던 패키지 구조를 레이어별로 통합한다

hyogenie-v committed:
refactor: 사용하지 않는 파라미터 삭제

1. 컨트롤러에서 사용되지 않는 파라미터를 삭제한다

hyogenie-v committed:
refactor: CommentController 클래스에서 @RestController 애너테이션을 사용하도록 변경

1. @ResponseBody 애너테이션을 반복적으로 사용하는 부분을 @RestController 애너테이션으로 통합한다

hyogenie-v committed:
refactor: 댓글 작성 및 수정 Controll 메서드가 엔티티가 아닌 DTO를 응답하도록 수정

hyogenie-v committed:
refactor: dto 패키지 위치를 service 내로 변경

hyogenie-v committed:
fix: 버그 수정

1. 댓글 작성 응답의 사용자 정보가 null로 받아지는 버그를 수정한다
   1-1. CommentService에서 Comment 객체를 repository로 직접 저장하도록 수정한다
2. 댓글 삭제 시 405 에러가 발생하는 버그를 수정한다
   2-1. CommentController를 RestController와 MvcController로 분리한다

hyogenie-v committed:
refactor: 댓글 관련 js 코드를 comment.js 파일로 추출

jihan805 reviewed (approved):
안녕하세요 에헴 ~!
미션 구현하느라 고생 많았어요 👍 👍 
머지 할게요 :)
